### ISSUE TITLE
Problem with case-sensitive chroot names in DNF5 copr plugin

### ISSUE THREAD

**Issue by FrostyX** [fedora-copr/copr#3539]
Reported on Matrix today.
Enabling a Copr project via DNF5 didn't work because of the capital F in the `Fedora-41-aarch64` chroot.

```
dnf copr enable ngompa/fedora-asahi-dev
Chroot not found in the given Copr project (Fedora-41-aarch64).
You can choose one of the available chroots explicitly:
...
fedora-41-aarch64
...
```

I am not sure how the capitalized chroot name came from. A workaround was

```
dnf copr enable ngompa/fedora-asahi-dev fedora-41-aarch64
```

**Comment by xsuchy** [comment]
I cannot reproduce on x86_64 on F41. May be aarch64 specific.

**Comment by FrostyX** [comment]
I tried in `fedora-41-aarch64`:

```
<mock-chroot> sh-5.2# python3
Python 3.13.0 (main, Oct  8 2024, 00:00:00) [GCC 14.2.1 20240912 (Red Hat 14.2.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import distro
>>> distro.id()
'fedora'
```

And it is correct too. I probably wouldn't search for the cause of the problem and only `.lower()` the chroot name

**Comment by nikromen** [comment]
matrix thread https://matrix.to/#/!rALCnapvdNTxUTuENl:matrix.org/$s-fyoKgj_lrWqabXtLvnMGXR1F9155qSHeeX3Pqmccg?via=fedora.im&via=matrix.org&via=mozilla.org

**Comment by PaddiM8** [comment]
It's probably Asahi related? I'm on Asahi too and noticed the same thing

**Comment by FrostyX** [comment]
Thank you very much for the confirmation @PaddiM8. We were suspecting that and wanted to ask some Asahi users that we know, to test this.

Can you please paste the output of these two commands?

```
python3 -c "import distro; print(distro.id())"
cat /etc/os-release |grep ID |grep -i fedora
```

**Comment by jannau** [comment]
```
python3 -c "import distro; print(distro.id())"
fedora-asahi-remix

cat /etc/os-release |grep ID |grep -i fedora
ID=fedora-asahi-remix
ID_LIKE=fedora
```

**Comment by PaddiM8** [comment]
```
~ ❯ python3 -c "import distro; print(distro.id())"
fedora-asahi-remix
~ ❯ cat /etc/os-release |grep ID |grep -i fedora
~ ❯
```

Upgraded from 39 to 41 just before noticing this btw

**Comment by FrostyX** [comment]
Thank you very much @jannau and @PaddiM8.

I am still confused how this happens. If I understand the DNF5 plugin correctly, it uses the ID variable from `/etc/os-release` that you showed us to be lowercase. And DNF4 plugin does some guessing magic, having several fallbacks. But in neither of them I can clearly see a problem.

Can you please let us know what DNF version are you using? So we can focus on debugging the right codebase.

```
dnf --version
```

In case, it is DNF4, these should hopefully help us locate the problem.

```
python3 -c "from distro import name, version, codename; print((name(), version(), codename()))"
python3 -c "from platform import linux_distribution; print(linux_distribution())"
cat /etc/os-release |grep -E "^NAME=|^VERSION_ID="
```

**Comment by jannau** [comment]
This is indeed an asahi specific issue. `fedora-asahi-remix-release-common` carries `/usr/share/dnf/plugins/copr.vendor.conf` with:
```
[main]
distribution = Fedora
releasever = 41
```
correcting the distribution to `fedora` fixes the issue. I'll see to get this fixed and this issue can be closed.

**Comment by FrostyX** [comment]
Great, thank you @jannau!

**Comment by FrostyX** [comment]
I proposed a fix in https://github.com/rpm-software-management/dnf5/pull/2561 . It should fix the issue but I don't have an actual Fedora Asahi system, so not 100% confirmed.

**Comment by FrostyX** [comment]
I decided to close the PR ^^ because the consensus was it should be fixed in the Asahi config. And it seems it was already fixed:

```
$ docker run -it fedora:43 bash
[root@c98369d19bb9 /]# dnf copr enable @asahi/fedora-remix-branding
[root@c98369d19bb9 /]# dnf install fedora-asahi-remix-release-common
[root@c98369d19bb9 /]# cat /usr/share/dnf/plugins/copr.vendor.conf |grep distribution
distribution = fedora
```

I think we can close the issue. Please reopen if it still doesn't work properly.

**Comment by jannau** [comment]
yes, we fixed the config at the end of 2024 with https://pagure.io/fedora-asahi/fedora-asahi-remix-release/pull-request/3 I just forgot to ping back here, sorry.
